crates/router/src/routes/app.rs

```diff
@@ -31,11 +33,11 @@ use crate::{
 pub struct AppState {
     pub flow_name: String,
     pub store: Box<dyn StorageInterface>,
-    pub conf: settings::Settings,
+    pub conf: Arc<settings::Settings>,
     #[cfg(feature = "email")]
     pub email_client: Box<dyn EmailClient>,
     #[cfg(feature = "kms")]
-    pub kms_secrets: settings::ActiveKmsSecrets,
+    pub kms_secrets: Arc<settings::ActiveKmsSecrets>,
     pub api_client: Box<dyn crate::services::ApiClient>,
```

Comment:
is arc needed for the other things as well ?

does clone on box do a deep clone...

Reply:
Converting `email _client` to `Arc` .